api/common/modelstatus.go

```diff
@@ -37,8 +45,40 @@ func (c *ModelStatusAPI) ModelStatus(tags ...names.ModelTag) ([]base.ModelStatus
 		return nil, err
 	}
 
-	results := make([]base.ModelStatus, len(result.Results))
-	for i, r := range result.Results {
+	if olderVersion {
+		// This caters for the fact that older version of base.ModelStatus
+		// did not have an Error field.
+		return c.processModelStatusResultsV3(result.Results)
+	}
+
+	return c.processModelStatusResults(result.Results)
+}
+
+func (c *ModelStatusAPI) processModelStatusResults(rs []params.ModelStatus) ([]base.ModelStatus, error) {
+	results := make([]base.ModelStatus, len(rs))
+	for i, r := range rs {
+		if r.Error != nil {
+			results[i].Error = errors.Trace(r.Error)
+			continue
+		}
+		model, err := names.ParseModelTag(r.ModelTag)
+		if err != nil {
+			results[i].Error = errors.Annotatef(err, "model tag %v", r.ModelTag)
```

Comment:
this will result in a message like
model tag foo: "foo" is not a valid model tag

no need for the annotate

Reply:
The annotation was there before. But i have indeed been wondering how much stuttering it would generate. I'll remove annotation :)